src/SeleniumLibrary/keywords/expectedconditions.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2016-     Robot Framework Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from SeleniumLibrary.base import LibraryComponent, keyword
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+class ExpectedConditionKeywords(LibraryComponent):
+    @keyword
+    def wait_for_expected_condition(self, condition, *args):
+        wait = WebDriverWait(self.driver, 10, 0.1)
+        # import sys,pdb;pdb.Pdb(stdout=sys.__stdout__).set_trace()
+        c = getattr(EC, condition)
+        wait.until(c(*args))
```

Comment:
I've been looking through the expectedconditions documentation.
Depending on the expectedcondition it seems either a bool, webelement or string is returned.
Do we want to add these return values?  (probably yes)
This also adds verification possibilities.

Reply:
I lean towards Yes but I also see there are a lot of different return values and combinations. So I am not sure what that means for properly setting the return. I also wonder if it is possible to set the return values based upon which expected condition is called?

I want to recommend that we put this into the next release and get expected conditions out there. Thoughts?

Reply:
I think for now the "wait for" is basically a boolean already, and can be used as such. You also show this in the tests. So for now it is ok, and maybe at a later stage look at what is needed?

Reply:
Ok, so I think we could set the type for the wait for expected condition keyword as you are right. That is singular in focus and has a return type. I was concerned about setting the return type on the (underlying) expected condition which seems a bit problematic or complicated based upon which expected condition it was. This was something i didn't want to hold up a release on given the amount of time and my lack of seeing how this can be done with multiple different method search with their own return type.

Now noting that the wait is a type and can be added now, looking at the selenium code I am confused about the type [[L81](https://github.com/SeleniumHQ/selenium/blob/trunk/py/selenium/webdriver/support/wait.py#L81)]. It looks like a generic type "T" with no really specifics on it. I would have thought it would have been Boolean as well. I don't know  if this is generic return type because Selenium didn't have a need to specify itself. Or if i am just understand the  type hinting here. I will note there is the similar until_not method in the same file which has an addition type `Literal[True]` as well. Still this doesn't help me to understand this but at least it is different.

Reply:
I'm not familiar enough with typing to say this with full certainty, but to me it seems like:
"until" waits until the response is no longer False or Falsy, so for conditions where a boolean is returned this is obvious. For responses with a WebElement or anything else it's basically the same; no response is still Falsy/False, but when a response is received it becomes True/Truthy so the "until" succeeds. No specific type needed, and the return value could be anything.

Reply:
Going to leave this as is for now ..